### PR TITLE
LINK-1648 | 3/3 Signups page

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
Page to list and filter the signups of the registration

This one of the 3 PRs that will implement signups page. PRs are splitted to following parts:
- MenuDropdown component (https://github.com/City-of-Helsinki/linkedregistrations-ui/pull/95)
- Attendance list page action buttons (https://github.com/City-of-Helsinki/linkedregistrations-ui/pull/96)
- **Signups page**

## Partially closes
[LINK-1648](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1648)

## Screenshot
Screenshot 2023-10-30 at 11 12 19

[LINK-1648]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1648?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: public/locales/en/signups.json [hunk 1]
```diff
@@ -0,0 +1,19 @@
+{
+  "attendeeStatus": {
+    "attending": "Attending",
+    "waitlisted": "In wait list"
```
Comment (harriris-vincit) [inline public/locales/en/signups.json:4]:
```suggestion
    "waitlisted": "In waiting list"
```
I believe this should be the more common/natural term

File: src/domain/singups/signupsTable/SignupsTable.tsx [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    </div>
+  );
+};
+
+const EmailColumn: FC<ColumnProps> = ({ signup }) => {
+  const { email } = getSignupFields({ signup });
+
+  return (
+    <>
+      {email ??
```
[the remark this answers is not in the document]
Comment (jorilindell) [reply src/domain/singups/signupsTable/SignupsTable.tsx]: Good catch. This is not intentional at all. Changed it to `{email || "-"}`. Fixed the similar issue for the phone number as well

File: public/locales/en/signups.json (hunk 1 above)
Comment (jorilindell) [reply public/locales/en/signups.json]: Changed

**Comment by harriris-vincit** [comment]
@jorilindell In my local environment I seem to be missing at least two signups that both belong to two different signup groups, but that belong to the same registration. These two don't show up in the signups table at all. I've been trying to figure out why that is but unfortunately with little luck. Each page after the first one seem to be showing different signups than what the back-end returns.

**Comment by jorilindell** [comment]
> @jorilindell In my local environment I seem to be missing at least two signups that both belong to two different signup groups, but that belong to the same registration. These two don't show up in the signups table at all. I've been trying to figure out why that is but unfortunately with little luck. Each page after the first one seem to be showing different signups than what the back-end returns.

The table is showing only signups with `attendee_status =. attending`. Could this be the reason?

**Comment by harriris-vincit** [comment]
> > @jorilindell In my local environment I seem to be missing at least two signups that both belong to two different signup groups, but that belong to the same registration. These two don't show up in the signups table at all. I've been trying to figure out why that is but unfortunately with little luck. Each page after the first one seem to be showing different signups than what the back-end returns.
> 
> The table is showing only signups with `attendee_status =. attending`. Could this be the reason?

Unfortunately not as both the missing signups have that status too. If I do a text search, those two show up in the UI table. Same thing if I modify the page_size to be large enough: both show up in the single long page.

**Comment by jorilindell** [comment]
> > > @jorilindell In my local environment I seem to be missing at least two signups that both belong to two different signup groups, but that belong to the same registration. These two don't show up in the signups table at all. I've been trying to figure out why that is but unfortunately with little luck. Each page after the first one seem to be showing different signups than what the back-end returns.
> > 
> > 
> > The table is showing only signups with `attendee_status =. attending`. Could this be the reason?
> 
> Unfortunately not as both the missing signups have that status too. If I do a text search, those two show up in the UI table. Same thing if I modify the page_size to be large enough: both show up in the single long page.

Does the API call (`/signup/?attendee_status=attending&page=1&page_size=10&registration=44&text=` or similar) return all the expected signups?

**Comment by harriris-vincit** [comment]
> > > > @jorilindell In my local environment I seem to be missing at least two signups that both belong to two different signup groups, but that belong to the same registration. These two don't show up in the signups table at all. I've been trying to figure out why that is but unfortunately with little luck. Each page after the first one seem to be showing different signups than what the back-end returns.
> > > 
> > > 
> > > The table is showing only signups with `attendee_status =. attending`. Could this be the reason?
> > 
> > 
> > Unfortunately not as both the missing signups have that status too. If I do a text search, those two show up in the UI table. Same thing if I modify the page_size to be large enough: both show up in the single long page.
> 
> @harriris-vincit Does the API call (`/signup/?attendee_status=attending&page=1&page_size=10&registration=44&text=` or similar) return all the expected signups?

Yes it does, both when the UI does it and when I do the same query with Postman. The first page is the same, but then it gets different when page is something else than 1.

I just noticed that the back-end passes an ordered queryset to `paginate_queryset` when the user is not a superuser, but when the user is a superuser the queryset is unordered. This then results in the different page contents while the un-paged qs actually contains the same signups otherwise

**Review by harriris-vincit — APPROVED** [review]
LGTM - the discussed issue with some signups not showing up when testing is an issue in the back-end code. A fix has been made and is in the PR phase currently.

**Pull request merged**